File: dataParse/pdfParse.py
import json
from pdfminer.high_level import extract_text

def extract_and_save_as_json(file_path, output_path):
    text = extract_text(file_path)
    data = {
        "file": file_path,
        "content": text
    }
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=4)

file_path = r"C:\home\tonghao\paper\2212.09297.pdf"
output_path = "output.json"
extract_and_save_as_json(file_path, output_path)
print(f"Saved JSON to {output_path}")

File: dataParse/pdfParse.js
const pdf2json = require("pdf2json");
const fs = require('fs');
const pdf = require('pdf-parse');
// const puppeteer = require("puppeteer");
// const pdfjsLib = require("pdfjs-dist");

const fileRoute = `C:\\home\\tonghao\\paper\\2212.09297.pdf`

// function parsePdf() {
//     const pdfParser = new pdf2json();
//     pdfParser.on("pdfParser_dataReady", pdfData => {
//         const result = JSON.stringify(pdfData);
//         console.log(result);
//     });
//
//     pdfParser.loadPDF(fileRoute);
// }
//
// parsePdf();

// 读取 PDF 文件
const dataBuffer = fs.readFileSync(fileRoute);

// 解析 PDF
pdf(dataBuffer).then(function(data) {
  // 输出 PDF 的元数据
  console.log(data.info); // PDF 的元数据
  console.log(data.text); // 提取的文本内容
});